plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/M2RepositoryService.java

```diff
@@ -109,4 +109,16 @@ protected void activate() {
   void setMavenHelper(MavenHelper mavenHelper) {
     this.mavenHelper = mavenHelper;
   }
+
+  /**
+   * First checks if the artifact is available locally. If not, then it tries to resolve it via
+   * Maven.
+   */
+  @Override
+  public void checkArtifactAvailable(LibraryFile libraryFile, IProgressMonitor monitor) throws CoreException {
```

Comment:
If it's trying to resolve it, then it might be better named _makeArtifactAvailable_?